lib/dfdutils/printdfd.c

```diff
@@ -15,10 +15,663 @@
  * Author: Andrew Garrard
  */
 
+#include <assert.h>
+#include <stdbool.h>
 #include <stdio.h>
+#include <string.h>
 #include <KHR/khr_df.h>
 #include "dfd.h"
 
+
+const char* dfdToStringVendorID(khr_df_vendorid_e value) {
+    switch (value) {
+    case KHR_DF_VENDORID_KHRONOS:
+        return "KHR_DF_VENDORID_KHRONOS";
+
+    case KHR_DF_VENDORID_MAX:
+        // These enum values are not meant for string representation. Ignore
+        break;
+    }
+    return NULL;
+}
+
+const char* dfdToStringDescriptorType(khr_df_khr_descriptortype_e value) {
+    switch (value) {
+    case KHR_DF_KHR_DESCRIPTORTYPE_BASICFORMAT:
+        return "KHR_DF_KHR_DESCRIPTORTYPE_BASICFORMAT";
+    case KHR_DF_KHR_DESCRIPTORTYPE_ADDITIONAL_PLANES:
+        return "KHR_DF_KHR_DESCRIPTORTYPE_ADDITIONAL_PLANES";
+    case KHR_DF_KHR_DESCRIPTORTYPE_ADDITIONAL_DIMENSIONS:
+        return "KHR_DF_KHR_DESCRIPTORTYPE_ADDITIONAL_DIMENSIONS";
+
+    case KHR_DF_KHR_DESCRIPTORTYPE_NEEDED_FOR_WRITE_BIT:
+    case KHR_DF_KHR_DESCRIPTORTYPE_NEEDED_FOR_DECODE_BIT:
+    case KHR_DF_KHR_DESCRIPTORTYPE_MAX:
+        // These enum values are not meant for string representation. Ignore
+        break;
+    }
+    return NULL;
+}
+
+const char* dfdToStringVersionNumber(khr_df_versionnumber_e value) {
+    switch (value) {
+    case KHR_DF_VERSIONNUMBER_1_1:
+    // case KHR_DF_VERSIONNUMBER_1_0: // Fallthrough, Matching values
+        return "KHR_DF_VERSIONNUMBER_1_1";
+    case KHR_DF_VERSIONNUMBER_1_2:
+        return "KHR_DF_VERSIONNUMBER_1_2";
+    case KHR_DF_VERSIONNUMBER_1_3:
+        return "KHR_DF_VERSIONNUMBER_1_3";
+
+    // case KHR_DF_VERSIONNUMBER_LATEST: // Fallthrough, Matching values
+    case KHR_DF_VERSIONNUMBER_MAX:
+        // These enum values are not meant for string representation. Ignore
+        break;
+    }
+    return NULL;
+}
+
+const char* dfdToStringFlagsBit(uint32_t bit_index, bool bit_value) {
+    switch (bit_index) {
+    case 0:
+        return bit_value ? "KHR_DF_FLAG_ALPHA_PREMULTIPLIED" : "KHR_DF_FLAG_ALPHA_STRAIGHT";
+    default:
+        return NULL;
+    }
+}
+
+const char* dfdToStringTransferFunction(khr_df_transfer_e value) {
+    switch (value) {
+    case KHR_DF_TRANSFER_UNSPECIFIED:
+        return "KHR_DF_TRANSFER_UNSPECIFIED";
+    case KHR_DF_TRANSFER_LINEAR:
+        return "KHR_DF_TRANSFER_LINEAR";
+    case KHR_DF_TRANSFER_SRGB:
+        return "KHR_DF_TRANSFER_SRGB";
+    case KHR_DF_TRANSFER_ITU:
+        return "KHR_DF_TRANSFER_ITU";
+    case KHR_DF_TRANSFER_NTSC:
+    // case KHR_DF_TRANSFER_SMTPE170M: // Fallthrough, Matching values
+        return "KHR_DF_TRANSFER_NTSC";
```

Comment:
For text output only I am fine with doing it, but this will make a little mess for the to string functions where I have to branch based on text/json representation.

Please note that there are sometimes 3 names for the same value.
The full list of colliding enum values are (with the currently selected options marked):

KHR_DF_VERSIONNUMBER_1_0 or KHR_DF_VERSIONNUMBER_1_1 (Selected)
KHR_DF_MODEL_DXT1A or KHR_DF_MODEL_BC1A (Selected)
KHR_DF_MODEL_DXT2 or KHR_DF_MODEL_DXT3 or KHR_DF_MODEL_BC2 (Selected)
KHR_DF_MODEL_DXT4 or KHR_DF_MODEL_DXT5 or KHR_DF_MODEL_BC3 (Selected)
KHR_DF_TRANSFER_ITU (Selected) or KHR_DF_TRANSFER_SMTPE170M
KHR_DF_PRIMARIES_BT709 or KHR_DF_PRIMARIES_SRGB (Selected)
KHR_DF_CHANNEL_BC6H_COLOR (selected) or KHR_DF_CHANNEL_BC6H_DATA
KHR_DF_CHANNEL_BC7_COLOR (selected) or KHR_DF_CHANNEL_BC7_DATA
KHR_DF_CHANNEL_ETC1_COLOR (selected) or KHR_DF_CHANNEL_ETC1_DATA
KHR_DF_CHANNEL_PVRTC_COLOR (selected) or KHR_DF_CHANNEL_PVRTC_DATA
KHR_DF_CHANNEL_PVRTC2_COLOR (selected) or KHR_DF_CHANNEL_PVRTC2_DATA

Reply:
OK, let's keep it as-is; most of these should definitely be printed with only one value. For primaries, I think BT.709 is a better choice than sRGB.

Reply:
Changing to KHR_DF_PRIMARIES_BT709. Keeping everything else as is.